test/unit_test/offset_tree_tests.cc

```diff
@@ -47,6 +49,27 @@ using predictions_t = vector<pair<float, float>>;
 using scores_t = std::vector<float>;
 
 void predict_test_helper(const predictions_t& base_reduction_predictions, const scores_t& expected_scores);
+test_learner_t* get_test_harness_reduction(const predictions_t& base_reduction_predictions);
+
+BOOST_AUTO_TEST_CASE(offset_tree_learn_basic){
+  // Setup a test harness base reduction
+  const auto test_harness = get_test_harness_reduction({
+      {.9, .1},
+      {.9, .1}
+  });
+
+  VW::offset_tree::offset_tree tree;
+  tree.init(3);
+  example ec;
+  ec.pred.a_s = v_init<ACTION_SCORE::action_score>();
+  ec.l.cb = CB::label();
+  ec.l.cb.costs.push_back({-1.0,1,0.5 ,0.0});
+  
+  tree.learn(*as_singleline(test_harness), ec);
+  
+  destroy_free<test_learner_t>(test_harness);
```

Comment:
Why can't we simply call delete?

Reply:
This is allocated following standard VW idiom ie. calloc().  The proper way to dealloc is free().